### PR TITLE
add multi-project chapters and crd reference generator

### PR DESCRIPTION
**Pull request by avinal**
## Changes

- add support for multi-project chapters
- add crd reference generator using https://github.com/giantswarm/crd-docs-generator
- update GitHub Actions to use the new crd-docs-generator
- add missing CSS rules
- add template for linking relative markdown files

Signed-off-by: Avinal Kumar <avinal@redhat.com>

## Test Deployment
https://avinal.space/kcp-dev.github.io/docs/

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -45,8 +46,10 @@ jobs:
       - run: npm install
       - run: npm ci
 
-      - name: Build
-        run: hugo --minify
+      - name: Generate and Build
```
Comment (sttts) [inline .github/workflows/gh-pages.yml]: make it another step

File: scripts/crd-ref-gen/run-crd-ref-gen.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+CRD_DOCS_GENERATOR_VERSION=0.10.0
+# set destination to crd-reference directory in the docs
+DESTINATION="/content/en/docs/kcp-core/crd-reference"
+
+# Clear output folder
+find "${PWD}/${DESTINATION}" -type f -not -name "_index.md" | xargs -I '{}' rm '{}'
+
+# Generate new content
+docker run --rm \
```
Comment (sttts) [inline scripts/crd-ref-gen/run-crd-ref-gen.sh]: this works in actions?

Comment (avinal) [reply scripts/crd-ref-gen/run-crd-ref-gen.sh]: I saw actions docs, docker is already set up in GitHub Actions. let me check again.

Comment (avinal) [reply scripts/crd-ref-gen/run-crd-ref-gen.sh]: https://github.com/actions/runner-images/blob/26563d7f30529963c7c1f7491d657c55f5e1a818/images/linux/Ubuntu2204-Readme.md?plain=1#L98

Podman is already installed so I will change the command

**Review by sttts — APPROVED** [review]

**Review by sttts — APPROVED** [review]

**Pull request merged**